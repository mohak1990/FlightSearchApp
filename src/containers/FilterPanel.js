import React from 'react';
import FilterPanel from '../components/filterPanel'
import { connect } from 'react-redux'
import {  } from '../modules/filterPanel'

class FilterPanelContainer extends React.Component {

  render() {
    return (
        <FilterPanel />
    );
  }
}

const mapStateToProps = (state, ownProps) => ({

})

const mapDispatchToProps = function(dispatch, ownProps){

  return {

  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(FilterPanelContainer)
