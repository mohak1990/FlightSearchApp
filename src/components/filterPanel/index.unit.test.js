import React from 'react';
import FilterPanel from './index';
import { mount, shallow } from 'enzyme';
import ShallowRenderer from 'react-test-renderer/shallow'

test('Snapshot testing for FilterPanel component', () => {
  const renderer = new ShallowRenderer()
  const result = renderer.render(<FilterPanel />)
  expect(result).toMatchSnapshot()
});

it('render FilterPanel correctly', () => {
    const FilterPanelComponent = mount(<FilterPanel onSearch={()=>{}} />).render();
    expect(FilterPanelComponent.hasClass('filter_panel')).toBeTruthy();
});

it('check if FilterPanel component exist', () => {
    const FilterPanelComponent = shallow(<FilterPanel onSearch={()=>{}} />);
    expect(FilterPanelComponent.find('Tabs')).toHaveLength(1);
});
