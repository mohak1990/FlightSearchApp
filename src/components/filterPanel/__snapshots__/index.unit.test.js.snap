// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing for FilterPanel component 1`] = `
<div
  className="filter_panel"
>
  <TabsComponent>
    <div
      label="One Way"
    >
      <div>
        <AutoCompleteComponent
          className="filter_panel--element"
          list={
            Array [
              Object {
                "key": "pun",
                "value": "Pune (PNQ)",
              },
              Object {
                "key": "bom",
                "value": "Mumbai (BOM)",
              },
              Object {
                "key": "blr",
                "value": "Bengaluru (BLR)",
              },
              Object {
                "key": "del",
                "value": "Delhi (DEL)",
              },
            ]
          }
          name="origin-city"
          onClick={[Function]}
          placeholder="Enter Origin City"
          type="text"
        />
      </div>
      <div>
        <AutoCompleteComponent
          className="filter_panel--element"
          list={
            Array [
              Object {
                "key": "pun",
                "value": "Pune (PNQ)",
              },
              Object {
                "key": "bom",
                "value": "Mumbai (BOM)",
              },
              Object {
                "key": "blr",
                "value": "Bengaluru (BLR)",
              },
              Object {
                "key": "del",
                "value": "Delhi (DEL)",
              },
            ]
          }
          name="dest-city"
          onClick={[Function]}
          placeholder="Enter Destination City"
          type="text"
        />
      </div>
      <div>
        <DatePicker
          allowSameDay={false}
          className="filter_panel--element"
          dateFormat="MM/dd/yyyy"
          dateFormatCalendar="LLLL yyyy"
          disabled={false}
          disabledKeyboardNavigation={false}
          dropdownMode="scroll"
          minDate={2019-01-20T20:08:53.227Z}
          monthsShown={1}
          nextMonthButtonLabel="Next month"
          onBlur={[Function]}
          onChange={[Function]}
          onClickOutside={[Function]}
          onFocus={[Function]}
          onInputClick={[Function]}
          onInputError={[Function]}
          onKeyDown={[Function]}
          onMonthChange={[Function]}
          onSelect={[Function]}
          onYearChange={[Function]}
          placeholderText="MM/DD/YYYY"
          preventOpenOnFocus={false}
          previousMonthButtonLabel="Previous Month"
          readOnly={false}
          renderDayContents={[Function]}
          selected={2019-01-20T20:08:53.193Z}
          shouldCloseOnSelect={true}
          showTimeSelect={false}
          timeCaption="Time"
          timeIntervals={30}
          withPortal={false}
        />
      </div>
      <div>
        <InputComponent
          className="filter_panel--element"
          name="passengers"
          options={Array []}
          placeholder="Select Passengers"
          type="select"
        />
      </div>
      <div>
        <InputComponent
          className="filter_panel--element filter_panel--element--button"
          disabled={true}
          name="search"
          type="button"
          value="Search"
        />
      </div>
    </div>
    <div
      label="Return"
    >
      <div>
        <AutoCompleteComponent
          className="filter_panel--element"
          list={
            Array [
              Object {
                "key": "pun",
                "value": "Pune (PNQ)",
              },
              Object {
                "key": "bom",
                "value": "Mumbai (BOM)",
              },
              Object {
                "key": "blr",
                "value": "Bengaluru (BLR)",
              },
              Object {
                "key": "del",
                "value": "Delhi (DEL)",
              },
            ]
          }
          name="origin-city"
          onClick={[Function]}
          placeholder="Enter Origin City"
          type="text"
        />
      </div>
      <div>
        <AutoCompleteComponent
          className="filter_panel--element"
          list={
            Array [
              Object {
                "key": "pun",
                "value": "Pune (PNQ)",
              },
              Object {
                "key": "bom",
                "value": "Mumbai (BOM)",
              },
              Object {
                "key": "blr",
                "value": "Bengaluru (BLR)",
              },
              Object {
                "key": "del",
                "value": "Delhi (DEL)",
              },
            ]
          }
          name="dest-city"
          onClick={[Function]}
          placeholder="Enter Destination City"
          type="text"
        />
      </div>
      <div>
        <DatePicker
          allowSameDay={false}
          className="filter_panel--element"
          dateFormat="MM/dd/yyyy"
          dateFormatCalendar="LLLL yyyy"
          disabled={false}
          disabledKeyboardNavigation={false}
          dropdownMode="scroll"
          minDate={2019-01-20T20:08:53.227Z}
          monthsShown={1}
          nextMonthButtonLabel="Next month"
          onBlur={[Function]}
          onChange={[Function]}
          onClickOutside={[Function]}
          onFocus={[Function]}
          onInputClick={[Function]}
          onInputError={[Function]}
          onKeyDown={[Function]}
          onMonthChange={[Function]}
          onSelect={[Function]}
          onYearChange={[Function]}
          placeholderText="MM/DD/YYYY"
          preventOpenOnFocus={false}
          previousMonthButtonLabel="Previous Month"
          readOnly={false}
          renderDayContents={[Function]}
          selected={2019-01-20T20:08:53.193Z}
          shouldCloseOnSelect={true}
          showTimeSelect={false}
          timeCaption="Time"
          timeIntervals={30}
          withPortal={false}
        />
      </div>
      <div>
        <DatePicker
          allowSameDay={false}
          className="filter_panel--element"
          dateFormat="MM/dd/yyyy"
          dateFormatCalendar="LLLL yyyy"
          disabled={false}
          disabledKeyboardNavigation={false}
          dropdownMode="scroll"
          minDate={2019-01-21T20:08:53.193Z}
          monthsShown={1}
          nextMonthButtonLabel="Next month"
          onBlur={[Function]}
          onChange={[Function]}
          onClickOutside={[Function]}
          onFocus={[Function]}
          onInputClick={[Function]}
          onInputError={[Function]}
          onKeyDown={[Function]}
          onMonthChange={[Function]}
          onSelect={[Function]}
          onYearChange={[Function]}
          placeholderText="MM/DD/YYYY"
          preventOpenOnFocus={false}
          previousMonthButtonLabel="Previous Month"
          readOnly={false}
          renderDayContents={[Function]}
          selected={2019-01-20T20:08:53.193Z}
          shouldCloseOnSelect={true}
          showTimeSelect={false}
          timeCaption="Time"
          timeIntervals={30}
          withPortal={false}
        />
      </div>
      <div>
        <InputComponent
          className="filter_panel--element"
          name="passengers"
          options={Array []}
          placeholder="Select Passengers"
          type="select"
        />
      </div>
      <div>
        <InputComponent
          className="filter_panel--element filter_panel--element--button"
          disabled={true}
          name="search"
          type="button"
          value="Search"
        />
      </div>
    </div>
  </TabsComponent>
</div>
`;
